debezium-core/src/main/java/io/debezium/relational/topic/TopicMapper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational.topic;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+import io.debezium.relational.Table;
+
+import java.util.Map;
+
+/**
+ * @author David Leibovic
+ */
+public abstract class TopicMapper {
+
+    protected String topicPrefix;
+    protected Table table;
+
+    public TopicMapper setTopicPrefix(String topicPrefix) {
+        this.topicPrefix = topicPrefix;
+        return this;
+    }
+
+    public TopicMapper setTable(Table table) {
```

Comment:
So I think it's a bit odd that a single `TopicMapper` instance is used within a connector like MySQL and yet it's stateful, meaning that should a connector interleave calls to the same instance then the state gets messed up. Sure, MySQL and PostgreSQL don't need that, but shared stateful instances place constraints on the connectors that a stateless object wouldn't. Am I missing something?

Have you thought about trying to make this API stateless?